sql/tools/toMongo.js

```diff
@@ -0,0 +1,36 @@
+import db from '../sequelize';
+import { User as MongoUser } from '../../server/models/user.model';
+
+const SqlUser = db.User;
+const SqlDevice = db.Device;
+
+(async () => {
+    const LIMIT = 1000;
+    let offset = 0;
+    let users = [];
+    do {
+        users = (await SqlUser.findAll({
+            offset,
+            limit: LIMIT,
+            include: [{
+                model: SqlDevice,
+            }],
+        })).map(user => ({
+            uuid: user.uuid,
+            username: user.username,
+            devices: user.Devices.map(device => ({
+                type: device.type,
+                token: device.token,
+                status: device.disabled ? 'disabled' : 'enabled',
+            })),
+        }));
+        await MongoUser.insertMany(users);
```

Comment:
looking at this again: I could collect all the `insertMany` promises in an array and then `await Promise.all(insertPromises)` after the do-while loop.

Thoughts?

Reply:
It probably won't affect any of our actual migrations though, because we only have a couple and they all have < 1000 users probably?

Reply:
Meh. Even if we had 1,000,000 users, this loop would only have to run 1,000 times. This is a migration, rather than a function that runs as a part of the normal application's behavior, so running this loop 1,000 times is not a problem. Note also that running the loop 1,000 times might not even take that long.

That said, I think it's ok the way it is.